sensorprocess/sensorprocess.go

```diff
@@ -115,3 +145,150 @@ func tryAddLidarReading(ctx context.Context, reading []byte, readingTime time.Ti
 	timeElapsedMs := int(time.Since(startTime).Milliseconds())
 	return int(math.Max(0, float64(config.LidarDataRateMsec-timeElapsedMs)))
 }
+
+// StartIMU polls the IMU to get the next sensor reading and adds it to the cartofacade.
+// stops when the context is Done.
+func (config *Config) StartIMU(
+	ctx context.Context,
+) bool {
+	for {
+		select {
+		case <-ctx.Done():
+			return false
+		default:
+			if config.stopped {
+				return true
+			}
+			if jobDone := config.addIMUReading(ctx); jobDone {
+				return true
+			}
+		}
+	}
+}
```

Comment:
I think it might be a bit cleaner to put this in Start so that we do not have to maintain stopping multiple processes

Reply:
I looked into doing it in one loop. While offline mode will work well, the issue is live mode. We would need to use tickers vs the sleeps, and the integration tests now produce differing results (this final one may be caused by `VIAM_CARTO_UNABLE_TO_ACQUIRE_LOCK` error, which I would need to look into more):

https://github.com/jeremyrhyde/viam-cartographer/pull/1

I think sticking with the two data loops for now ends up producing a cleaner setup, happy to talk about this more offline if you feel differently

Reply:
I see what you mean and agree - sorry for sending you down a rabbit hole!

Reply:
Do we throw away IMU readings that occur after the last lidar reading?

Reply:
We want to stop adding imu data after lidar data has been complete for two reasons:
1. For optimization runs after lidar data has been complete https://github.com/viamrobotics/viam-cartographer/pull/254#discussion_r1304608372
2. Once we are no longer at getting lidar data, our constraint equations for loop closure would only be relying on imu data which may lead to errors. The IMU constraint equations are supposed to provide additional data, not be the sole arbitrator of pose estimation.